cpp/models/ode_secirvvs/model.h

```diff
@@ -543,21 +543,45 @@ class Simulation : public BaseT
     {
     }
 
-    void apply_b161(double t)
+    using ParamT = CustomIndexArray<UncertainValue, AgeGroup>;
+
+    /**
+    * @brief Applies the effect of a new variant of a disease to the transmission probability of the model.
+    * 
+    * This function adjusts the transmission probability of the disease for each age group based on the share of the new variant.
+    * The share of the new variant is calculated based on the time `t` and the start day of the new variant.
+    * The transmission probability is then updated for each age group in the model.
+    * 
+    * Based on Equation (35) and (36) in doi.org/10.1371/journal.pcbi.1010054
+    * 
+    * @param [in] t The current time.
+    * @param [in] base_infectiousness The base infectiousness of the old variant for each age group.
+    */
+
+    void apply_variant(const double t, const ParamT base_infectiousness)
     {
+        auto start_day             = this->get_model().parameters.template get<StartDay>();
+        auto start_day_new_variant = this->get_model().parameters.template get<StartDayNewVariant>();
 
-        auto start_day   = this->get_model().parameters.template get<StartDay>();
-        auto b161_growth = (start_day - get_day_in_year(Date(2021, 6, 6))) * 0.1666667;
-        // 2 equal to the share of the delta variant on June 6
-        double share_new_variant = std::min(1.0, pow(2, t * 0.1666667 + b161_growth) * 0.01);
-        size_t num_groups        = (size_t)this->get_model().parameters.get_num_groups();
-        for (size_t i = 0; i < num_groups; ++i) {
-            double new_transmission =
-                (1 - share_new_variant) *
-                    this->get_model().parameters.template get<BaseInfectiousnessB117>()[(AgeGroup)i] +
-                share_new_variant * this->get_model().parameters.template get<BaseInfectiousnessB161>()[(AgeGroup)i];
-            this->get_model().parameters.template get<TransmissionProbabilityOnContact>()[(AgeGroup)i] =
-                new_transmission;
+        if (start_day + t >= start_day_new_variant) {
```

Comment:
Your previous code looked more like you'd want to use '>' in this if, are you sure '>='  is correct here?

Reply:
The idea here is as follows. `start_day `and `start_day_new_variant `are both integers. `t` is of type double. We call `t` for time points 0.5, 1, 1.5 and so on. Since `t` is a double, it can happen that the "if condition" is not called, although we actually want it, because of rounding errors. Therefore, I have now added the tolerance again. And the >= is intentional :)

Reply:
Thanks for the input :) 
The function is called semi-discretely at fixed points in the model-specific advance and is not part of the ODE integration. So we can assume that we won't have this problem here :)